lib/ansible/plugins/strategy/__init__.py

```diff
@@ -315,6 +320,27 @@ def _queue_task(self, host, task, task_vars, play_context):
             queued = False
             starting_worker = self._cur_worker
             while True:
+                # Determine the "rewind point" of the worker list. This means we start
+                # iterating over the list of workers until the end of the list is found.
+                # Normally, that is simply the length of the workers list (as determined
+                # by the forks or serial setting), however a task/block/play may "throttle"
+                # that limit down.
+                templar = Templar(loader=self._loader, variables=task_vars)
+                try:
+                    throttle = int(templar.template(task.throttle))
+                except Exception as e:
+                    raise AnsibleError("Failed to convert the throttle value to an integer.", obj=task._ds, orig_exc=e)
+
+                rewind_point = len(self._workers)
```

Comment:
Is there some other way other than doing a len on `self._workers`?  With forking and your threading work this should be reliable, but is it possible that 3rd party process plugins in the future would not have a static length list?  Maybe they pop and append, and this could catch at a point where it's not at the max.

Reply:
In the future, it could read the property from the process manager plugin. This will all have to be folded in there so I'm not overly concerned about this right now.